services/replicator/outconn.go

```diff
@@ -128,116 +135,63 @@ func (conn *outConnection) writeCreditsStream() {
+	defer close(conn.msgsCh)
+	defer conn.wg.Done()
 
-	var sealMsgRead bool
 	var numMsgsRead int32
 
 	// Note we must continue read until we hit an error before returning from this function
 	// Because the websocket client only tear down the underlying connection when it gets a read error
-readloop:
 	for {
 		rmc, err := conn.stream.Read()
 		if err != nil {
 			conn.logger.WithField(common.TagErr, err).Error(`Error reading msg`)
-			go conn.close()
 			return
 		}
 
-		switch rmc.GetType() {
-		case store.ReadMessageContentType_MESSAGE:
-			msg := rmc.GetMessage()
-
-			if sealMsgRead {
-				conn.logger.WithFields(bark.Fields{
-					"seqNum": msg.Message.GetSequenceNumber(),
-				}).Error("regular message read after seal message")
-				go conn.close()
-				continue readloop
-			}
-
-			// Sequence number check to make sure we get monotonically increasing sequence number.
-			if lastSeqNum+1 != msg.Message.GetSequenceNumber() && lastSeqNum != -1 {
-				expectedSeqNum := 1 + lastSeqNum
-
-				conn.logger.WithFields(bark.Fields{
-					"seqNum":         msg.Message.GetSequenceNumber(),
-					"expectedSeqNum": expectedSeqNum,
-				}).Error("sequence number out of order")
-				go conn.close()
-				continue readloop
-			}
-
-			// update the lastSeqNum to this value
-			lastSeqNum = msg.Message.GetSequenceNumber()
-
-			conn.m3Client.IncCounter(conn.metricsScope, metrics.ReplicatorOutConnMsgRead)
-
-			// now push msg to the msg channel (which will in turn be pushed to client)
-			// Note this is a blocking call here
-			select {
-			case conn.msgsCh <- rmc:
-				numMsgsRead++
-				atomic.AddInt32(&conn.totalMsgReplicated, 1)
-				atomic.StoreInt64(&conn.lastMsgReplicatedTime, time.Now().UnixNano())
-			case <-conn.closeChannel:
-				conn.logger.Info(`writing msg to the channel failed because of shutdown`)
-				continue readloop
-			}
-
-		case store.ReadMessageContentType_SEALED:
-			seal := rmc.GetSealed()
-			conn.logger.WithField(`SequenceNumber`, seal.GetSequenceNumber()).Info(`extent sealed`)
-			sealMsgRead = true
-
-			// now push msg to the msg channel (which will in turn be pushed to client)
-			// Note this is a blocking call here
-			select {
-			case conn.msgsCh <- rmc:
-				numMsgsRead++
-				atomic.AddInt32(&conn.totalMsgReplicated, 1)
-				atomic.StoreInt64(&conn.lastMsgReplicatedTime, time.Now().UnixNano())
-			case <-conn.closeChannel:
-				conn.logger.Info(`writing msg to the channel failed because of shutdown`)
-			}
-
-			continue readloop
-
-		case store.ReadMessageContentType_ERROR:
-			msgErr := rmc.GetError()
-			conn.logger.WithField(`Message`, msgErr.GetMessage()).Error(`received error from reading msg`)
-			go conn.close()
-			continue readloop
+		conn.m3Client.IncCounter(conn.metricsScope, metrics.ReplicatorOutConnMsgRead)
+		if rmc.GetType() == store.ReadMessageContentType_SEALED {
+			conn.logger.WithField(`SequenceNumber`, rmc.GetSealed().GetSequenceNumber()).Info(`extent sealed`)
```

Comment:
for debug/assertion purposes .. do you want to remember that you saw a "sealed" message .. and in case you don't an error from the next read, log an error or something?